docs/DeveloperGuide.md

```diff
@@ -154,6 +154,89 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Find Command
+
+The find command allows a user to specify all the different fields which they want to search by.
+It is implemented by parsing all the different fields that the user is searching with, and creating the relevant predicate for each field.
+For example:
+* Searching by name `n/NAME` will create a NameContainsKeywordsPredicate
+* Searching by role `r/ROLE` will create a RoleContainsKeywordsPredicate
+* Searching by overtime hours with a 'less than' comparison `o/<OVERTIME` will create a OvertimeLessThanPredicate
+
+and so on.
+
+Each field the user specifies is located and parsed with the `tokenize` function from ArgumentTokenizer.
+The ArgumentTokenizer splits the user input into key-value pairs of the field's prefix and the values to query with, which can then be handled by the FindCommandParser.
+Each predicate is then generated by passing the value found in the ArgumentMultimap created by ArgumentTokenizer.
+In general, there are 2 main types of predicates.
+
+1. Keyword matching predicates
+    * These predicates check for a match between the user's query and the respective field in each `Person` in HeRon
+    * The keywords are easily found from the ArgumentMultimap and are parsed using the `String#split()` method.
+
+    One possible issue is that keywords containing spaces cannot be used. Therefore, a user input such as "Admin Assistant" will match "HR Assistant" and "IT Admin". 
+    However, the current implementation allows for more flexible searching
+   
+2. Comparison based predicates
+    * These predicates compare their given value with the respective field in each `Person` in HeRon.
+        * For example, a LeaveLessThanPredicate will check if the `Person` has a number of leaves strictly less than the given value.
+    * First, the respective method found in `FindCommandParser` for creating a comparison predicate is called.
+        * For example, to get a `SalaryIsLessThanPredicate`, `FindCommandParser#getSalaryComparisonPredicate` is called with the user's input.
+    * The comparison type is found using the `parseComparator()` method in FindCommandParser, which returns a CompareType, an enum found in FindCommandParser which has 5 values representing the different available comparisons: "more than", "more than or equal", "less than", "less than or equal", and "equal".
+    * The value to compare to is then found using the `getComparisonValue()` method in FindCommandParser, which returns a String to be parsed.
+        * This is done because the different fields take in different number types. For example, the Salary field takes in a `float`, whereas the Leave field takes in an `int`.
```

Comment:
I think maybe you have to change 'Salary' here to 'HourlySalary'? And also i think that it takes in a 'double' rather than 'float'?